test/SlowTests/Server/Documents/ETL/Olap/RavenDB_16311.cs

```diff
@@ -0,0 +1,122 @@
+                            Company = $"companies/{i}"
+                        });
+                    }
+
+                    for (int i = 0; i < 28; i++)
+                    {
+                        await session.StoreAsync(new Query.Order
+                        {
+                            Id = $"orders/{i + 31}",
+                            OrderedAt = baseline.AddMonths(1).AddDays(i),
+                            ShipVia = $"shippers/{i + 31}",
+                            Company = $"companies/{i + 31}"
+                        });
+                    }
+
+                    await session.SaveChangesAsync();
+                }
+
+                var database = await GetDatabase(store.Database);
+
+                using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                {
+                    var result = (OlapEtlTestScriptResult)OlapEtl.TestScript(new TestOlapEtlScript
+                    {
+                        DocumentId = "orders/1",
+                        Configuration = new OlapEtlConfiguration()
+                        {
+                            Name = "simulate",
+                            Transforms =
+                                {
+                                    new Transformation()
+                                    {
+                                        Collections =
+                                        {
+                                            "Orders"
+                                        },
+                                        Name = "MonthlyOrders",
+                                        Script =
+                                            @"
+                                                var orderDate = new Date(this.OrderedAt);
+                                                var year = orderDate.getFullYear();
+                                                var month = orderDate.getMonth();
+                                                var key = new Date(year, month);
+
+                                                output('test output')
+
+                                                loadToOrders(partitionBy([['order_date', key]]),
+                                                    {
+                                                        Company : this.Company,
+                                                        ShipVia : this.ShipVia
+                                                    });
+                                                "
+                                    }
+                                }
+                        }
+                    }, database, database.ServerStore, context);
+
+                    Assert.Equal(0, result.TransformationErrors.Count);
+
+                    Assert.Equal(1, result.ItemsByPartition.Count);
+
+                    Assert.Equal(4, result.ItemsByPartition[0].Columns.Count);
+
```

Comment:
maybe also assert that `ItemsByPartition[0].Key ` = `Orders/order_date=2020-01-01` ?

Reply:
Done. Although it is "Orders/order_date=2020-01-01-00-00"